src/NServiceBus.Core/DataBus/DataBus.cs

```diff
@@ -1,44 +1,27 @@
 namespace NServiceBus.Features
 {
-    using System;
     using System.Threading.Tasks;
     using NServiceBus.DataBus;
-    using Settings;
 
     /// <summary>
     /// Used to configure the databus.
     /// </summary>
-    public class DataBus : Feature
+    public class DataBus : Feature, IRequireService<DataBusStorage>
     {
-        internal DataBus()
-        {
-            Defaults(s => s.EnableFeatureByDefault(GetSelectedFeatureForDataBus(s)));
-        }
-
-        static Type GetSelectedFeatureForDataBus(SettingsHolder settings)
-        {
-            DataBusDefinition dataBusDefinition;
-
-            if (!settings.TryGet("SelectedDataBus", out dataBusDefinition))
-            {
-                dataBusDefinition = new FileShareDataBus();
-            }
-
-            return dataBusDefinition.ProvidedByFeature();
-        }
-
         /// <summary>
         /// Called when the features is activated.
         /// </summary>
         protected internal override void Setup(FeatureConfigurationContext context)
         {
+            //todo: this can be a service dependency instead
             if (!context.Container.HasComponent<IDataBusSerializer>())
             {
                 context.Container.ConfigureComponent<DefaultDataBusSerializer>(DependencyLifecycle.SingleInstance);
             }
 
-            context.RegisterStartupTask(b => b.Build<IDataBusInitializer>());
-            context.Container.ConfigureComponent<IDataBusInitializer>(DependencyLifecycle.SingleInstance);
+            var storage = context.GetService<DataBusStorage>().Storage;
```

Comment:
Me and @SzymonPobiega had the crazy idea to pass `DataBusStorage` as a ctor dependency instead.

We could do this if we get rid of the need to have a parameter less ctor on features